mongo.go

```diff
@@ -0,0 +1,473 @@
+package mongods
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	dsq "github.com/ipfs/go-datastore/query"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/jbenet/goprocess"
+	dsextensions "github.com/textileio/go-datastore-extensions"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	ErrClosed = errors.New("datastore was closed")
+
+	log = logging.Logger("mongods")
+)
+
+type MongoDS struct {
+	m          *mongo.Client
+	db         *mongo.Database
+	col        *mongo.Collection
+	opTimeout  time.Duration
+	txnTimeout time.Duration
+
+	lock   sync.RWMutex
+	closed bool
+}
+
+var _ datastore.Datastore = (*MongoDS)(nil)
+var _ datastore.Batching = (*MongoDS)(nil)
+var _ datastore.TxnDatastore = (*MongoDS)(nil)
+var _ dsextensions.ExtendedDatastore = (*MongoDS)(nil)
+
+type keyValue struct {
+	Key   string `bson:"_id"`
+	Value []byte `bson:"v"`
+}
+
+func New(ctx context.Context, uri string, dbName string, collName string, opts ...Option) (*MongoDS, error) {
```

Comment:
Here's an interesting subtle thing. I made the `db` and `collName` configurable. 
This possibly allows to multiplex different applications in the same MongoDB instance.

That's to say: the `Datastore` way of thinking is having namespacing at the Key level. That would be in the defined `(dbName, collName)` pair. We could in theory use the same `dbName` and another collection and have a new namespace for another application. 

Food for thought.

Reply:
Cool, makes sense. I could see `dbName` and `collName` being options with defaults, since unless you're planning to use multiple namespaces/apps/etc, you probably won't care much what these values are.

Reply:
Agree, I'll make them options with reasonable defaults  👍🏼

Reply:
@sanderpick , I did this change. Maybe you want to checkout the last commit of this PR if makes sense to you?

Reply:
Looks good!